src/ds++/VectorView.hh

```diff
@@ -0,0 +1,138 @@
+//----------------------------------*-C++-*----------------------------------//
+/*!
+ * \file   ds++/VectorView.hh
+ * \author Kelly G. Thompson <kgt@lanl.gov>, Steve Nolen <sdnolen@lanl.gov>
+ * \date   Friday, Dec 13, 2019, 17:49 pm
+ * \brief  A view class that presents contiguous 1-D as multi-D.
+ * \note   Copyright (C) 2019 Triad National Security, LLC.
+ *         All rights reserved. */
+//---------------------------------------------------------------------------//
+
+#ifndef rtt_dsxx_VectorView_hh
+#define rtt_dsxx_VectorView_hh
+
+#include "Assert.hh"
+#include <array>
+#include <vector>
+
+namespace rtt_dsxx {
+
+//============================================================================//
+/*!
+ * \class VectorView
+ * \brief A wrapper for std::vector<T> that presents a compatible multi-D
+ *        'view'.
+ *
+ * \tparam T The data storage for the underlying std::vector<T>.
+ * \tparam Extents A variatic template list of integral values that represent
+ *         the max value of each dimension of the 'view'.
+ * \todo Add operator= to allow assignment operations.
+ *
+ * Example:
+ *
+ * \code
+ * vector<double> v1 = { 1,2,3,4,5,6,7,8,9 };
+ * VectorView<double,3,3,3> myvecview(v1);
```

Comment:
Does VectorView require that sizes be known at compile time?

Reply:
Sizes do need to be known at compile time, according to my test on compiler explorer.

Reply:
That limits the applicability of this feature for my own uses. Do others still find this useful?